scripts/mbedtls_dev/c_build_helper.py

```diff
@@ -128,18 +131,32 @@ def get_c_expression_values(
         )
         c_file.close()
         cc = os.getenv('CC', 'cc')
-        cc_is_msvc = os.path.split(cc)[1].lower() in ('cl', 'cl.exe')
         cmd = [cc]
+
+        global _cc_is_msvc #pylint: disable=global-statement,invalid-name
```

Comment:
Caching `_cc_is_msvc` is fragile: maybe at some point we'll use this function in a script that does builds with multiple compilers. I don't plan to do this anywhere soon, we can live with caching, but please add a note to this function's documentation.

Reply:
Why bother with caching anyway? I've been treating the CC calls as an expensive operation. `test_psa_constant_names.py` only calls this function once. `generate_psa_tests.py` currently calls this function multiple times, but that's only because of a bug in its own caching mechanism which #4385 is fixing. So I think we shouldn't bother with caching here at all.

Reply:
Ok, in that case, I'll remove the caching.

Reply:
Note: if we were doing caching, I think I'd be more inclined to use the facilities of [functools](https://docs.python.org/3/library/functools.html) than rolling our own (with the associated pylint warnings to silence).